auslib/test/admin/views/test_rules.py

```diff
@@ -141,9 +141,22 @@ def testNewRuleWithoutPermission(self):
     def testMissingFields(self):
         # But we still need to pass product, because permission checking
         # is done before what we're testing
-        ret = self._post('/rules', data=dict({'product': 'a'}))
+        post_data = {'product': 'a'}
+        ret = self._post('/rules', data=post_data)
+        self.assertEquals(ret.status_code, 400, "Status Code: %d, Data: %s" % (ret.status_code, ret.data))
+        # Connexion and manual request.json validation methods will preempt and throw 400 error as
+        # soon as the first field invalidates. Only one field may be present in the response.data depending
+        # upon the order of fields sorted in swagger yaml file and in views.
+        self.assertTrue('title' in ret.data, msg=ret.data)
```

Comment:
The request object passed [here](https://github.com/zalando/connexion/blob/master/connexion/decorators/validation.py#L105) is an instance of [ConnexionRequest](https://github.com/zalando/connexion/blob/master/connexion/lifecycle.py#L2). Here's the [link](https://pastebin.mozilla.org/9021108) debug log when I post a new rule.

When validate method is invoked [here](https://github.com/zalando/connexion/blob/master/connexion/decorators/validation.py#L133), then the entire json data is validated in one go based on schema. Hence , jsonschema validation throws the first error it encounters and the message cannot be customized unless we edit jsonschema code. Also the ConnexionRequest class objects have no instance variable or field which can identify the invalid field's key or parameter name. Ex: 

> {
  "detail": "u'mindosr' is not one of ['major', 'minor']",
  "status": 400,
  "title": "Bad Request",
  "type": "about:blank"
}

The above response doesn't contain the key name that update_type field value is invalid. However if it is a 'required' property and is missing then the detail key says "'update_type' field is missing".